File: packages/core/src/readiness/ReadinessScore.test.ts
import { describe, it, expect } from "vitest";
import { MobileAIReadiness } from "./MobileAIReadiness";

describe("ReadinessScore", () => {
  it("should calculate overall score from dimensions", () => {
    const readiness = new MobileAIReadiness();
    const score = readiness.calculateReadiness({
      contextEnabled: true,
      memoryEnabled: true,
      privacyEnabled: true,
      actionsEnabled: true,
      offlineSupport: "partial",
      uxContinuity: "good",
      appStoreReadiness: "review_needed",
    });

    expect(score.overall).toBeGreaterThan(0);
    expect(score.overall).toBeLessThanOrEqual(100);
  });

  it("should have all 7 dimensions", () => {
    const readiness = new MobileAIReadiness();
    const score = readiness.calculateReadiness({
      contextEnabled: true,
      memoryEnabled: true,
      privacyEnabled: true,
      actionsEnabled: true,
      offlineSupport: "full",
      uxContinuity: "excellent",
      appStoreReadiness: "ready",
    });

    expect(score.dimensions.contextAwareness).toBeDefined();
    expect(score.dimensions.memoryQuality).toBeDefined();
    expect(score.dimensions.privacySafety).toBeDefined();
    expect(score.dimensions.actionReliability).toBeDefined();
    expect(score.dimensions.offlineResilience).toBeDefined();
    expect(score.dimensions.uxContinuity).toBeDefined();
    expect(score.dimensions.appStoreReadiness).toBeDefined();
  });

  it("should provide recommendations", () => {
    const readiness = new MobileAIReadiness();
    const score = readiness.calculateReadiness({
      contextEnabled: false,
      memoryEnabled: false,
      privacyEnabled: false,
      actionsEnabled: false,
      offlineSupport: "none",
      uxContinuity: "poor",
      appStoreReadiness: "not_ready",
    });

    expect(score.dimensions.contextAwareness.recommendations.length).toBeGreaterThan(0);
    expect(score.dimensions.privacySafety.recommendations.length).toBeGreaterThan(0);
  });

  it("should have higher score when all features enabled", () => {
    const readiness = new MobileAIReadiness();
    
    const lowScore = readiness.calculateReadiness({
      contextEnabled: false,
      memoryEnabled: false,
      privacyEnabled: false,
      actionsEnabled: false,
      offlineSupport: "none",
      uxContinuity: "poor",
      appStoreReadiness: "not_ready",
    });

    const highScore = readiness.calculateReadiness({
      contextEnabled: true,
      memoryEnabled: true,
      privacyEnabled: true,
      actionsEnabled: true,
      offlineSupport: "full",
      uxContinuity: "excellent",
      appStoreReadiness: "ready",
    });

    expect(highScore.overall).toBeGreaterThan(lowScore.overall);
  });
});

File: packages/compose/src/test/kotlin/ai/mobileagentos/compose/ReadinessScoreTest.kt
package ai.mobileagentos.compose

import ai.mobileagentos.compose.readiness.*
import org.junit.Assert.*
import org.junit.Test

class ReadinessScoreTest {
    
    @Test
    fun `calculate readiness score`() {
        val dimensions = mapOf(
            ReadinessDimension.CONTEXT_AWARENESS to DimensionScore(80),
            ReadinessDimension.MEMORY_QUALITY to DimensionScore(75),
            ReadinessDimension.PRIVACY_SAFETY to DimensionScore(85),
            ReadinessDimension.ACTION_RELIABILITY to DimensionScore(70),
            ReadinessDimension.OFFLINE_RESILIENCE to DimensionScore(60),
            ReadinessDimension.UX_CONTINUITY to DimensionScore(75),
            ReadinessDimension.APP_STORE_READINESS to DimensionScore(60)
        )
        
        val score = ReadinessScore.calculate(dimensions)
        
        assertTrue(score.overall > 0)
        assertTrue(score.overall <= 100)
        assertEquals(dimensions.size, score.dimensions.size)
    }
    
    @Test
    fun `readiness tier classification`() {
        val notReady = ReadinessScore(35, emptyMap(), "")
        assertEquals(ReadinessTier.NOT_READY, notReady.tier)
        
        val needsWork = ReadinessScore(55, emptyMap(), "")
        assertEquals(ReadinessTier.NEEDS_WORK, needsWork.tier)
        
        val good = ReadinessScore(75, emptyMap(), "")
        assertEquals(ReadinessTier.GOOD, good.tier)
        
        val excellent = ReadinessScore(90, emptyMap(), "")
        assertEquals(ReadinessTier.EXCELLENT, excellent.tier)
    }
    
    @Test
    fun `MobileAIReadiness calculates correctly`() {
        val readiness = MobileAIReadiness()
        
        val input = ReadinessInput(
            contextEnabled = true,
            memoryEnabled = true,
            privacyEnabled = true,
            actionsEnabled = true
        )
        
        val score = readiness.calculateReadiness(input)
        
        assertTrue(score.overall > 50)
        assertEquals(7, score.dimensions.size)
    }
}